src/main/resources/utilities/populate_default_locus_range_for_species.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import configparser
+import urllib.parse
+
+from pymongo import MongoClient
+from __init__ import *
+
+DEFAULT_LOCUS_RANGE_COLLECTION_NAME = "defaultLocusRange"
+VARIANT_COLLECTION_NAME = "variants_2_0"
+VARIANT_COLLECTION_NAME_ALT = "variants_1_2"
+
+# This is basically the MongoDB pipeline version of
+# select top 1 from
+#  (select chr, start/1e6, count(*) as num_entries from variants_2_0 group by 1,2) counts order by num_entries desc
+pick_locus_range_query = [
+        {
+            "$group": {
+                "_id": {
+                    "chr": "$chr",
+                    "start_1M_multiple": {"$trunc": {"$divide": ["$start", 1e6]}}
+                },
+                "COUNT(*)": {
+                    "$sum": 1
+                }
+            }
+        },
+        {
+            "$project": {
+                "chr": "$_id.chr",
+                "start_1M_multiple": "$_id.start_1M_multiple",
+                "num_entries": "$COUNT(*)",
+                "_id": 0
+            }
+        },
+        {
+            "$sort": {"num_entries": -1}
+        },
+        {
+            "$group": {
+                "_id": None,
+                "topChrStartPick": {"$first": "$$ROOT"}
```

Comment:
Are the 2 last stages making something different than `..., {"$sort": {"num_entries": -1}}, {"$limit:1}]`?

Reply:
I was not sure if $limit guarantees ordering or it retrieves just one entry at random so I opted for $first. But looks like limit() does honor the order. so I will change this....

Reply:
yep, also a great optimization will be done:

> When a $sort precedes a $limit and there are no intervening stages that modify the number of documents, the optimizer can coalesce the $limit into the $sort. This allows the $sort operation to only maintain the top n results as it progresses, where n is the specified limit, and ensures that MongoDB only needs to store n items in memory. This optimization still applies when allowDiskUse is true and the n items exceed the aggregation memory limit.